src/saveload/afterload.cpp

```diff
@@ -733,6 +734,11 @@ bool AfterLoadGame()
 	 * must be done before loading sprites as some newgrfs check it */
 	TimerGameCalendar::SetDate(TimerGameCalendar::date, TimerGameCalendar::date_fract);
 
+	/* Determine economy dates based on calendar date */
+	if (IsSavegameVersionBefore(SLV_ECONOMY_DATE)) {
+		TimerGameEconomy::SetDate(static_cast<int32_t>(TimerGameCalendar::date), TimerGameCalendar::date_fract);
+	}
```

Comment:
This should not be done conditionally.

In the savegame, only `date` is stored, not `year`. This is deduced on load again by this unconditional statement as on line 735.

Reply:
Ah, yes. It needs to be conditional so that we always set call SetDate() but if no economy date is stored, we use the calendar date instead of the economy date.